Remove VAP manifests from Kueue configuration

ugiordan opened the pull request:
## Description

Remove Validation Admission Policy (VAP) manifests and related code from Kueue configuration since VAP functionality is now handled by a dedicated webhook implementation.

This change eliminates redundant VAP resources that were previously managed by the Kueue controller, streamlining the configuration and avoiding conflicts with the new webhook-based validation approach.

Additionally, implement direct cleanup of deprecated ValidatingAdmissionPolicyBinding resources during operator upgrades to ensure clean transitions from previous versions.

**JIRA**: [RHOAIENG-26500](https://issues.redhat.com/browse/RHOAIENG-26500)

### Changes Made:
- Remove entire `config/kueue-configs/ocp-4.17-addons` directory containing:
  * `validating_admission_policy.yaml`
  * `validating_admission_policy_binding.yaml` 
  * `kustomization.yaml`
- Remove VAP ownership and watching from kueue controller
- Remove `customizeResources` function (was only handling VAPB)
- Remove VAP GVK definitions from `pkg/cluster/gvk/gvk.go`
- Remove VAP RBAC permissions from:
  * `internal/controller/datasciencecluster/kubebuilder_rbac.go`
  * `config/rbac/role.yaml`
  * `bundle/manifests/opendatahub-operator.clusterserviceversion.yaml`
- Remove `ValidateKueueVAPReady` test function and related test case
- Fix typo in test function name: `ValidateKueueUnmanagedWhitoutOcpKueueOperator` → `ValidateKueueUnmanagedWithoutOcpKueueOperator`
- Add `cleanupDeprecatedVAPB` function to remove orphaned ValidatingAdmissionPolicyBinding resources during upgrades
- Add comprehensive unit tests for the cleanup functionality
- Use typed ValidatingAdmissionPolicyBinding struct for better type safety

## How Has This Been Tested?

- **Build Verification**: Confirmed all code compiles successfully with `go build ./internal/controller/components/kueue/...` and `go build ./cmd/main.go`
- **Manifest Generation**: Verified `make manifests bundle` generates correct RBAC without VAP permissions
- **Unit Testing**: New cleanup functionality tested with comprehensive unit tests:
  ```bash
  go test ./internal/controller/components/kueue/... -v -run TestCleanupDeprecatedVAPB
  ```
  - Tests cover both resource exists and resource not found scenarios
  - Tests validate proper error handling and logging
  - All tests pass successfully
- **E2E Testing**: Can be tested with Kueue component e2e tests:
  ```bash
  go test -v ./tests/e2e -run TestOdhOperator -timeout 30m --test-webhook=false --test-components=true --test-component=kueue
  ```

## Screenshot or short clip

N/A - Infrastructure/configuration change with no UI impact

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

zdtsw commented on internal/controller/components/kueue/kueue_controller.go:
```diff
@@ -57,14 +56,7 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
-		b = b.WatchesGVK(gvk.ValidatingAdmissionPolicyBinding)
-
-		// add OCP 4.17.0 specific menifests
+		// add OCP 4.17.0 specific manifests
```
but we still wanna call an emptry kustomization.yaml https://github.com/opendatahub-io/opendatahub-operator/blob/e6f0acebfd8c4d825b5afeb1c3567c3238f320e9/config/kueue-configs/ocp-4.17-addons/kustomization.yaml ?

lburgazzoli replied:
do it

zdtsw commented:
assume during upgrade:
- VAP will be removed due to gc
- VAPB, will this be removed as well? it does not have ownerreference by Kueue

ugiordan commented:
> assume during upgrade:
> 
> * VAP will be removed due to gc
> * VAPB, will this be removed as well? it does not have ownerreference by Kueue

@zdtsw Good point. I’ve added the upgrade cleanup logic here: https://github.com/opendatahub-io/opendatahub-operator/pull/2113/commits/c9a126d063c311ed6eac5c9c16d074ed1d96b519.

@lburgazzoli, I noticed there’s a `TODO` on the centralized `CleanupExistingResource` function, so I wasn’t sure if the upgrade cleanup logic should go there, or if we’d prefer an action-based approach, which seems cleaner to me. Happy to adjust based on whatever direction you think makes the most sense, or defer this if we’d rather handle it later since it’s not urgent.

ugiordan commented:
/hold working on action-based upgrade logic

ugiordan commented:
> > assume during upgrade:
> > 
> > * VAP will be removed due to gc
> > * VAPB, will this be removed as well? it does not have ownerreference by Kueue
> 
> @zdtsw Good point. I’ve added the upgrade cleanup logic here: [c9a126d](https://github.com/opendatahub-io/opendatahub-operator/commit/c9a126d063c311ed6eac5c9c16d074ed1d96b519).
> 
> @lburgazzoli, I noticed there’s a `TODO` on the centralized `CleanupExistingResource` function, so I wasn’t sure if the upgrade cleanup logic should go there, or if we’d prefer an action-based approach, which seems cleaner to me. Happy to adjust based on whatever direction you think makes the most sense, or defer this if we’d rather handle it later since it’s not urgent.

@lburgazzoli, @zdtsw, @valdar please share your feedback https://github.com/opendatahub-io/opendatahub-operator/pull/2113/commits/3a80df2e98bb8ea08677f423523b7a119e2b6f90

lburgazzoli commented on pkg/controller/actions/upgrade/action_upgrade.go:
```diff
@@ -0,0 +1,293 @@
+package upgrade
```
I'm not very sure about this action as:
1. I'd avoid using annotations, we could include release information to the components/services status and use it, which would avoid cluttering the annotations and also would avoid triggering spurious updates/reconciliation
2. For the current usage, the action seems a workaround for the [GC action limitation](https://github.com/opendatahub-io/opendatahub-operator/blob/main/pkg/controller/actions/gc/action_gc.go#L231-L253), so wonder if it would be better to find a way to improve the GC action instead of adding a new one. FWIW I was working on a more generic [cleanup action](https://github.com/lburgazzoli/opendatahub-operator/tree/cleanup-action) to replace the GC one, that could be of inspiration.
3. For the current usage, it is about deleting a single object, not sure it is worth adding an action for this. I understand that It may be useful for the future, but I'd defer the introduction of a new action for a later stage i.e. after having implemented/reasoned about the point 1 and 2 above

ugiordan commented on internal/controller/components/kueue/kueue_controller_actions.go:
```diff
@@ -200,3 +184,22 @@ func manageKueueAdminRoleBinding(ctx context.Context, rr *odhtypes.Reconciliatio
 }
+
+// Deprecated resource definitions for cleanup.
+var (
```
We don't need it anymore, and I'll replace it with the corresponding struct like
```
admissionregistrationv1 "k8s.io/api/admissionregistration/v1"

...
	vapb := &admissionregistrationv1.ValidatingAdmissionPolicyBinding{
		ObjectMeta: metav1.ObjectMeta{
			Name: "kueue-validating-admission-policy-binding",
		},
	}
```

ugiordan commented:
Reverted from the upgrade action framework as per @lburgazzoli's suggestion, since we'll work on it later.

Instead, implemented a normal cleanup action integrated directly into the existing upgrade infrastructure (`pkg/upgrade/upgrade.go`). This keeps the implementation simple while we develop the generic upgrade framework separately.

lburgazzoli commented on pkg/metadata/annotations/annotations.go:
```diff
@@ -17,6 +17,9 @@ const (
 const ManagementStateAnnotation = "component.opendatahub.io/management-state"
 
+// UpgradeCleanupAnnotationPrefix is used for marking upgrade cleanup completion per component per version.
+const UpgradeCleanupAnnotationPrefix = "platform.opendatahub.io/upgrade-cleanup"
```
probably not needed anymore

lburgazzoli commented:
@ugiordan need rebase/regen

ugiordan commented:
/test opendatahub-operator-e2e
